src/main/java/net/fabricmc/loom/LoomRepositoryPlugin.java

```diff
@@ -146,8 +147,8 @@ public static void setupForLegacyVersions(RepositoryHandler repositories) {
 		});
 	}
 
-	public static void forceLWJGLFromMavenCentral(RepositoryHandler repositories) {
-		if (repositories.findByName("MavenCentralLWJGL") != null) {
+	public static void forceLWJGLFromMavenCentral(RepositoryHandler repositories, LibraryContext context) {
+		if (repositories.findByName("MavenCentralLWJGL") != null && !context.usesLWJGL3()) {
```

Comment:
Are you sure this is needed? The related upgrade processors already handle this.

https://github.com/EssentialGG/architectury-loom/blob/89276b1713de6781fb5f67a559267f8a99043d1d/src/main/java/net/fabricmc/loom/configuration/providers/minecraft/library/processors/LWJGL3UpgradeLibraryProcessor.java#L48

https://github.com/EssentialGG/architectury-loom/blob/89276b1713de6781fb5f67a559267f8a99043d1d/src/main/java/net/fabricmc/loom/configuration/providers/minecraft/library/processors/ArmNativesLibraryProcessor.java#L50